diff --git a/plugins/pinmame/PinMAMEPlugin.cpp b/plugins/pinmame/PinMAMEPlugin.cpp
index 9ab3c7e3..71c61cad 100644
--- a/plugins/pinmame/PinMAMEPlugin.cpp
+++ b/plugins/pinmame/PinMAMEPlugin.cpp
@@ -474,8 +474,14 @@ MSGPI_EXPORT void MSGPIAPI PinMAMEPluginLoad(const uint32_t sessionId, const Msg
          pinmamePath = find_case_insensitive_directory_path(std::filesystem::path(vpxInfo.prefPath) / "pinmame"sv);
       }
       #elif defined(__APPLE__) || defined(__linux__)
+      if (pinmamePath.empty() && vpxApi != nullptr)
+      {
+         VPXInfo vpxInfo;
+         vpxApi->GetVpxInfo(&vpxInfo);
+         pinmamePath = std::filesystem::path(vpxInfo.prefPath) / "pinmame"sv;
+      }
       if (pinmamePath.empty())
          pinmamePath = std::filesystem::path(getenv("HOME")) / ".pinmame"sv;
       #endif
       // FIXME implement a last resort or just ask the user to define its path setup in the settings ?
