diff --git a/sources/src/fpp_softfloat.c b/sources/src/fpp_softfloat.c
index d2eb0c6..9a94c35 100644
--- a/sources/src/fpp_softfloat.c
+++ b/sources/src/fpp_softfloat.c
@@ -352,7 +352,10 @@ static void fp_getman(fpdata *a, fpdata *b)
 }
 static void fp_mod(fpdata *a, fpdata *b, uae_u64 *q, uae_u8 *s)
 {
-	a->fpx = floatx80_mod(a->fpx, b->fpx, q, s, &fs);
+	uint64_t quotient = *q;
+
+	a->fpx = floatx80_mod(a->fpx, b->fpx, &quotient, s, &fs);
+	*q = (uae_u64)quotient;
 }
 static void fp_sgldiv(fpdata *a, fpdata *b)
 {
@@ -364,7 +367,10 @@ static void fp_sglmul(fpdata *a, fpdata *b)
 }
 static void fp_rem(fpdata *a, fpdata *b, uae_u64 *q, uae_u8 *s)
 {
-	a->fpx = floatx80_rem(a->fpx, b->fpx, q, s, &fs);
+	uint64_t quotient = *q;
+
+	a->fpx = floatx80_rem(a->fpx, b->fpx, &quotient, s, &fs);
+	*q = (uae_u64)quotient;
 }
 static void fp_scale(fpdata *a, fpdata *b)
 {
