diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1b248099..de1f22f4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1200,13 +1200,13 @@ file(MAKE_DIRECTORY "${SHADNET_PROTO_OUT}")
 add_custom_command(
     OUTPUT
         "${SHADNET_PROTO_OUT}/shadnet.pb.cc"
         "${SHADNET_PROTO_OUT}/shadnet.pb.h"
-    COMMAND protobuf::protoc
+    COMMAND $<TARGET_FILE:protobuf::protoc>
         "--proto_path=${CMAKE_CURRENT_SOURCE_DIR}/src/shadnet"
         "--cpp_out=${SHADNET_PROTO_OUT}"
         "${CMAKE_CURRENT_SOURCE_DIR}/src/shadnet/shadnet.proto"
     DEPENDS
         "${CMAKE_CURRENT_SOURCE_DIR}/src/shadnet/shadnet.proto"
     COMMENT "Generating shadnet protobuf sources"
     VERBATIM
 )
