diff --git a/src/unix/osd_cpu.h b/src/unix/osd_cpu.h
index f7f4d4d..68d3a84 100644
--- a/src/unix/osd_cpu.h
+++ b/src/unix/osd_cpu.h
@@ -102,9 +102,11 @@ typedef union {

 /* disable BIG_SWITCH optimalisation in z80.c and m6809.c on buggy gcc
-   versions */
-#if (defined __GNUC__) && \
+   versions. GCC 13 for aarch64 can spend minutes optimizing m6809.c. */
+#if (defined __GNUC__) && defined(__aarch64__)
+#define BIG_SWITCH          0
+#elif (defined __GNUC__) && \
    ((__GNUC__ < 2) || ((__GNUC__ == 2) && (__GNUC_MINOR__ < 8)))
 #define BIG_SWITCH          0
 #else
 #define BIG_SWITCH          1
 #endif
