--- a/Sources/Jazz2/ContentResolver.cpp
+++ b/Sources/Jazz2/ContentResolver.cpp
@@ -240 +240 @@
-				_cachePath = fs::CombinePath(localStorage, "Jazz² Resurrection/Cache/"_s);
+				_cachePath = "/userdata/system/cache/jazz2/";
@@ -242,2 +242 @@
-					auto appData = fs::CombinePath(localStorage, NCINE_LINUX_PACKAGE);
-					_cachePath = fs::CombinePath(appData, "Cache/"_s);
+					fs::CreateDirectories(_cachePath);
@@ -249 +248 @@
-			_sourcePath = fs::CombinePath(fs::GetDirectoryName(_cachePath), "Source/"_s);
+			_sourcePath = "/userdata/roms/jazz2/";
--- a/Sources/Shared/IO/FileSystem.cpp
+++ b/Sources/Shared/IO/FileSystem.cpp
@@ -1385 +1385 @@
-			return CombinePath(config, applicationName);
+			return CombinePath(config, "jazz2");
