--- a/ThunkLibs/Generator/CMakeLists.txt	2026-04-12 13:45:06.435216556 +0300
+++ b/ThunkLibs/Generator/CMakeLists.txt	2026-04-12 13:45:12.911883221 +0300
@@ -1,5 +1,6 @@
 find_package(Clang REQUIRED CONFIG)
 find_package(LLVM REQUIRED CONFIG)
+find_package(fmt REQUIRED CONFIG)
 find_package(OpenSSL REQUIRED COMPONENTS Crypto)
 
 # Query clang's global resource directory for system include directories
@@ -14,13 +15,13 @@ endif()
 add_library(thunkgenlib STATIC analysis.cpp data_layout.cpp gen.cpp)
 target_include_directories(thunkgenlib INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
 target_include_directories(thunkgenlib SYSTEM PUBLIC ${CLANG_INCLUDE_DIRS} ${LLVM_INCLUDE_DIRS})
-target_link_libraries(thunkgenlib PUBLIC clang-cpp LLVM)
+target_link_libraries(thunkgenlib PUBLIC clangTooling clangFrontend clangAST LLVM)
 target_link_libraries(thunkgenlib PRIVATE OpenSSL::Crypto)
 target_link_libraries(thunkgenlib PRIVATE fmt::fmt)
 target_compile_definitions(thunkgenlib INTERFACE -DCLANG_RESOURCE_DIR="${CLANG_RESOURCE_DIR}")
 
 # Clang's libtooling won't compile with libstdc++'s debug mode
-target_compile_options(thunkgenlib PUBLIC "-U_GLIBCXX_DEBUG")
+target_compile_options(thunkgenlib PUBLIC "-U_GLIBCXX_DEBUG" "-fno-rtti")
 
 add_executable(thunkgen main.cpp)
 target_link_libraries(thunkgen PRIVATE thunkgenlib)
--- a/ThunkLibs/Generator/interface.h	2026-04-12 13:45:06.451883222 +0300
+++ b/ThunkLibs/Generator/interface.h	2026-04-12 13:45:12.915216555 +0300
@@ -1,4 +1,5 @@
 #include <clang/Tooling/Tooling.h>
+#include "data_layout.h"
 
 #include <optional>
 #include <string>
