--- a/ThunkLibs/GuestLibs/CMakeLists.txt
+++ b/ThunkLibs/GuestLibs/CMakeLists.txt
@@ -126,7 +126,7 @@ function(generate NAME SOURCE_FILE)
       # Expand compile definitions to space-separated list of -D parameters
       "$<$<BOOL:${compile_prop}>:;-D$<JOIN:${compile_prop},;-D>>"
       # Expand include directories to space-separated list of -isystem parameters
-      "$<$<BOOL:${prop}>:;-isystem$<JOIN:${prop},;-isystem>>"
+      "$<$<BOOL:${prop}>:;-isystem;$<JOIN:${prop},;-isystem;>>"
     VERBATIM
     COMMAND_EXPAND_LISTS)
