--- a/ThunkLibs/HostLibs/CMakeLists.txt
+++ b/ThunkLibs/HostLibs/CMakeLists.txt
@@ -143,7 +143,10 @@ foreach(GUEST_BITNESS IN LISTS BITNESS_LIST)
   add_host_lib(drm ${GUEST_BITNESS})
 endforeach()
 
-set(BITNESS_LIST "32;64")
+set(BITNESS_LIST "64")
+if (BUILD_32BIT_GUEST_THUNKS)
+  list(APPEND BITNESS_LIST "32")
+endif()
 foreach(GUEST_BITNESS IN LISTS BITNESS_LIST)
   if (BUILD_FEX_LINUX_TESTS)
     generate(libfex_thunk_test ${CMAKE_CURRENT_SOURCE_DIR}/../libfex_thunk_test/libfex_thunk_test_interface.cpp ${GUEST_BITNESS})
